types/src/shared/cache.ts

```diff
@@ -1,30 +1,55 @@
 import { redisClient } from "../shared/redis_client";
 
+// JSON-serializable primitive types.
+type JsonPrimitive = string | number | boolean | null;
```

Comment:
Pretty complex unfortunately... TS enforces strict checks when types are defined through interfaces and types...